source/common/conn_pool/conn_pool_base.h

```diff
@@ -65,6 +73,8 @@ class ActiveClient : public LinkedObject<ActiveClient>,
   // Returns the number of active streams on this connection.
   virtual uint32_t numActiveStreams() const PURE;
 
+  virtual bool hadNegativeDeltaOnStreamClosed() { return false; }
```

Comment:
nit: maybe leave a comment that h2 pool overrides this and h1 and tcp clients can not have negative concurrent stream count.